### PR TITLE
test: add Playwright

### PR DESCRIPTION
**Pull request by phoenix-ru**
### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation, readme or JSdoc annotations)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [x] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description
This PR adds Playwright E2E testing

### 📝 Checklist

- [ ] I have linked an issue or discussion.
- [x] I have added tests (if possible).
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: playground-local/app.vue [hunk 1]
```diff
@@ -10,31 +10,33 @@ const password = ref('')
 
 <template>
   <div>
-    <pre>Status: {{ status }}</pre>
+    <pre>Status: <span data-testid="status">{{ status }}</span></pre>
```
Comment (zoey-kaiser) [inline playground-local/app.vue:13]: Why do you name it `data-testid`? Would it make more sense to add another dash? -> `data-test-id`

File: playground-local/playwright.config.ts [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+ * See https://playwright.dev/docs/test-configuration.
+ */
+export default defineConfig({
+  testDir: './tests',
+  /* Run tests in files in parallel */
+  fullyParallel: true,
+  /* Fail the build on CI if you accidentally left test.only in the source code. */
+  forbidOnly: !!process.env.CI,
+  /* Retry on CI only */
+  retries: process.env.CI ? 2 : 0,
```
Comment (zoey-kaiser) [inline playground-local/playwright.config.ts:19]: So the CI will try twice before it completely fails. Is there a reason why we do not just let it run once?

File: playground-local/tests/local.spec.ts [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+const STATUS_AUTHENTICATED = 'authenticated'
+const STATUS_UNAUTHENTICATED = 'unauthenticated'
+
+describe('Local Provider', async () => {
+  await setup({
+    runner: 'vitest',
+    browser: true
+  })
+
+  test('it works', async () => {
```
Comment (zoey-kaiser) [inline playground-local/tests/local.spec.ts:14]: Do we want to improve this naming? What does "it works" mean?

File: playground-refresh/tests/refresh.spec.ts [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+const STATUS_AUTHENTICATED = 'authenticated'
+const STATUS_UNAUTHENTICATED = 'unauthenticated'
+
+describe('Refresh Provider', async () => {
+  await setup({
+    runner: 'vitest',
+    browser: true
+  })
+
+  test('it works', async () => {
```
Comment (zoey-kaiser) [inline playground-refresh/tests/refresh.spec.ts:14]: Same here for naming.

**Review by zoey-kaiser — CHANGES_REQUESTED** [review]

File: playground-local/playwright.config.ts (hunk 2 above)
Comment (phoenix-ru) [reply playground-local/playwright.config.ts:19]: This file is auto-generated by Playwright, I have not changed it

File: playground-local/tests/local.spec.ts (hunk 3 above)
Comment (phoenix-ru) [reply playground-local/tests/local.spec.ts:14]: What would you propose? I use that test name very often :)

Comment (phoenix-ru) [reply playground-local/tests/local.spec.ts:14]: Yeah, let's do it

**Review by zoey-kaiser — APPROVED** [review]

**Pull request merged**